packages/components/src/border-box-control/styles.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { css } from '@emotion/react';
+
+/**
+ * Internal dependencies
+ */
+import { COLORS, CONFIG } from '../utils';
+import { space } from '../ui/utils/space';
+
+export const BorderBoxControl = css``;
+
+export const LinkedBorderControl = css`
+	flex: 1;
+`;
+
+export const BorderBoxControlLinkedButton = css`
+	flex: 0;
+	flex-basis: 36px;
+	margin-top: 7px;
+`;
+
+export const BorderBoxControlVisualizer = css`
+	border: ${ CONFIG.borderWidth } solid ${ COLORS.gray[ 200 ] };
```

Comment:
I was initially a bit confused by how the visualizer behaves in a not-quite-WYSIWYG way, but thinking through it, I understood that it was because we need to support "partial" values, like `width` only. It's pretty clever, once I got it!

<img width="288" alt="Border width zero makes the visualizer border disappear" src="https://user-images.githubusercontent.com/555336/155982425-cd9a99b1-d605-46d5-9f03-5e3ce0eb25c7.png">
<img width="292" alt="Changing the border color changes the visualizer border color" src="https://user-images.githubusercontent.com/555336/155982506-e2c700c5-dac7-42d5-ba0b-36f5538881d7.png">
<img width="295" alt="Just setting the border width changes the visualizer border color to black (from a light gray)" src="https://user-images.githubusercontent.com/555336/155982586-23033053-67c0-4b51-9232-170d68c15c33.png">

---

Is there a reason I'm not allowed to set a `style` only?

<img src="https://user-images.githubusercontent.com/555336/155985556-27efe049-60d3-4767-87bc-45545e627988.gif" alt="CleanShot 2022-02-28 at 21 43 27" width="200">

Reply:
> I was initially a bit confused by how the visualizer behaves in a not-quite-WYSIWYG way, but thinking through it, I understood that it was because we need to support "partial" values, like `width` only. It's pretty clever, once I got it!

I've tried my best to make it intuitive. Reflecting a state where no selections have been made and inheritance is possible is a little tricky. 

As your screenshots show, when explicitly setting a zero width border we can reflect the fact there is no border. With only a color selected we do have a border just only with inherited width and style. Similarly, setting only the width to a non-zero number we need to handle an inherited color and style.

I'm definitely open to improvements. For now I'd be leaning to any improvements being handled via a separate PR.


> Is there a reason I'm not allowed to set a `style` only?

This came from other discussions around the border supports where border style was determined to be "secondary" to width and color. Part of that was that the style should be cleared whenever there is no border color or width. This in turn effectively means you can't set a border style only.

The `BorderControl` component itself has a prop that allows this sanitisation to be turned off. 

If it proves desirable, we could provide this on the `BorderBoxControl` as well in a follow-up.

Reply:
> I'm definitely open to improvements.

I think it's good! There are no logical inconsistencies, at the very least.

> This in turn effectively means you can't set a border style only.

I see. I'll just note that:

- In unlinked mode you can't set a border style only, but in linked mode you can.
- If a border style cannot be set for whatever reason, should that be communicated accessibly instead of silently ignoring the clicks? (Like by disabling the buttons?)

Feel free to move to a follow-up 🙂

Reply:
From memory, the reason you could set a style only on the linked control was that we might be dealing with mixed borders. For example, if the individual sides have different colors, styles and widths, then the linked control would have undefined values for those. Setting only a style here would then apply that to the individual sides.

https://user-images.githubusercontent.com/60436221/159430835-1dde7393-35ec-4f98-9a9a-34b12853f5be.mp4

I agree that we can refine this in a follow-up.